weave

```diff
@@ -938,6 +964,50 @@ dns_args() {
     fi
 }
 
+default_etc_hosts() {
+    NAME=$1
+    shift
+    echo "# created by Weave - BEGIN"
+    echo "# container hostname"
+    for CIDR in $ALL_CIDRS ; do
+        echo "${CIDR%/*}    $NAME"
+    done
+    echo
+    echo "# static names added with --add-host"
+    for EXTRA_HOST in "$@" ; do
+        NAME=$(echo $EXTRA_HOST | cut -f1 -d":")
+        IP=$(echo $EXTRA_HOST | cut -f2 -d":")
+        echo "$IP     $NAME"
+    done
+
+    cat <<-EOF
+
+# default localhost entries
+127.0.0.1       localhost
+::1             ip6-localhost ip6-loopback
+fe00::0         ip6-localnet
+ff00::0         ip6-mcastprefix
+ff02::1         ip6-allnodes
+ff02::2         ip6-allrouters
+EOF
+
+    echo "# created by Weave - END"
+}
+
+rewrite_etc_hosts() {
+    HOSTS_PATH_AND_NAME=$(docker inspect -f '{{.HostsPath}} {{.Config.Hostname}}' $CONTAINER)
+    HOSTS=$(echo $HOSTS_PATH_AND_NAME | cut -f1 -d" ")
+    NAME=$(echo $HOSTS_PATH_AND_NAME | cut -f2 -d" ")
```

Comment:
could reduce these three lines to

    eval $(docker inspect -f 'HOSTS={{.HostsPath}}; NAME={{.Config.Hostname}}' $CONTAINER)